backend-rust/src/indexer.rs

```diff
@@ -4084,23 +4188,38 @@ impl RestakeEarnings {
                             ELSE delegated_stake
                         END
                 WHERE address = $1
-                RETURNING index, delegated_restake_earnings",
+                RETURNING index, delegated_restake_earnings, delegated_target_baker_id",
             self.account_address.as_ref(),
             self.amount
         )
         .fetch_one(tx.as_mut())
         .await?;
-        // When delegated_restake_earnings is None the account is not delegating, so it
-        // might be baking.
-        if account_row.delegated_restake_earnings.is_none() {
+        if let Some(restake) = account_row.delegated_restake_earnings {
+            // Account is delegating.
+            if restake {
+                // Restake is enabled.
+                if let Some(pool) = account_row.delegated_target_baker_id {
+                    // delegating to a pool (and not the passive pool).
+                    sqlx::query!(
+                        "UPDATE bakers
+                             SET pool_total_staked = pool_total_staked + $2
+                         WHERE id = $1",
+                        pool,
+                        self.amount,
+                    )
+                    .execute(tx.as_mut())
+                    .await?;
+                }
+            }
```

Comment:
```suggestion
            if {
                if restake && let Some(pool) = account_row.delegated_target_baker_id {
                  // delegating to a pool (and not the passive pool).
                  sqlx::query!(
                      "UPDATE bakers
                           SET pool_total_staked = pool_total_staked + $2
                       WHERE id = $1",
                      pool,
                      self.amount,
                  )
                  .execute(tx.as_mut())
                  .await?;
            }
```

Reply:
`if <condition> && let ...` is not valid syntax in rust, but I can change it to:
```rust
if let (true, Some(pool)) = (restake, account_row.delegated_target_baker_id) {
    ...
}
```
Removes one indentation level, so maybe better :shrug: